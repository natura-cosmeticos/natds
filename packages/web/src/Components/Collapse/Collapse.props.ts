import { CollapseProps } from '@material-ui/core/Collapse'

export interface ICollapseProps extends CollapseProps {

  /**
   * The content of the component.
   *
   * @optional
   * @type node
   */
  children?: CollapseProps['children']

  /**
   * Override or extend the styles applied to the component.
   *
   * @optional
   * @type object
   */
  classes?: CollapseProps['classes']

  /**
   * The height of the container when collapsed.
   *
   * @default "0px"
   * @optional
   * @todo add support to size tokens
   * @type string | number
   */
  collapsedHeight?: CollapseProps['collapsedHeight']

  /**
   * The component used for the root node. Either a string to use a DOM element or a component.
   *
   * @default "div"
   * @optional
   * @type element
   */
  component?: CollapseProps['component']

  /**
   * If `true`, the component will transition in.
   *
   * @optional
   * @type bool
   */
  in?: CollapseProps['in']

  /**
   * The duration for the transition, in milliseconds.
   *
   * You may specify a single timeout for all transitions, or individually with an object:
   *
   * ```
   * { enter?: number, exit?: number }
   * ```
   *
   * Set to `auto` to automatically calculate transition time based on height.
   *
   * Defaults to `duration.standard`.
   *
   * @type number | object | "auto"
   */
  timeout?: CollapseProps['timeout']
}
