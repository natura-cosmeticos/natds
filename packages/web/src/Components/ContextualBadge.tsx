/* eslint-disable max-lines */
import React, { FunctionComponent, forwardRef } from "react";
import styled from "styled-components";
import { withTheme } from "@material-ui/core/styles";

import { IThemeWeb } from "../Themes";
import { getDefaultTheme } from "./shared";

const defaultValues = {
  lineHeight: 16,
};

type IContextualBadgeColors = "primary" | "secondary" | "info" | "error" | "warning" | "success" | "light" | "dark";

type IContextualBadgeTypes = "standard";

export interface IContextualBadgeProps {
  [propName: string]: any;

  /**
   * The input id property
   */
  id: string;

  /**
   * @optional
   */
  theme?: IThemeWeb | unknown;

  /**
   * @optional
   * Color of the ContextualBadge. Defaults to 'primary'
   */
  color?: IContextualBadgeColors;

  /**
   * @optional
   * Type of the ContextualBadge. Defaults to 'standard'
   */
  type?: IContextualBadgeTypes;

  /**
   * Classes to be used on the ContextualBadge
   */
  classes?: string;

  /**
   * Style to be used on the ContextualBadge
   */
  style?: React.CSSProperties;

  /**
   * Elements to be placed inside the ContextualBadge
   */
  children?: React.ReactNode;
}

// @todo refactor(web): refactor ContextualBadge component
export const ContextualBadge: FunctionComponent<IContextualBadgeProps> = forwardRef((
  props: IContextualBadgeProps,
  ref: any,
) => {
  const {
    children,
    classes,
    color = "primary",
    style,
    theme: providerTheme,
    type = "standard",
    ...rest
  } = props;

  const theme: any = React.useMemo(
    () => getDefaultTheme(providerTheme),
    [providerTheme],
  );

  const colorMap = React.useMemo(() => {
    const {
      primary, secondary, complementary, error, success, background, text,
    } = theme.palette;

    return {
      dark: {
        container: text.disabled,
        text: background.defaultContrastText,
      },
      error: {
        container: error.main,
        text: error.contrastText,
      },
      info: {
        container: complementary.link,
        text: complementary.linkContrastText,
      },
      light: {
        container: background.paper,
        text: background.paperContrastText,
      },
      primary: {
        container: primary.main,
        text: primary.contrastText,
      },
      secondary: {
        container: secondary.main,
        text: secondary.contrastText,
      },
      success: {
        container: success.main,
        text: success.contrastText,
      },
      warning: {
        container: complementary.warning,
        text: complementary.warningContrastText,
      },
    };
  }, [theme]);

  const fontMap = React.useMemo(() => {
    const { typography } = theme;

    return {
      fontFamily: typography.fontFamily,
      fontSize: typography.caption.fontSize,
      fontWeight: typography.caption.fontWeight,
      lineHeight: defaultValues.lineHeight,
    };
  }, [theme]);

  const constraints = React.useMemo(() => {
    const { spacing, shape } = theme;

    return {
      borderRadius: shape.badgeBorderRadius,
      padding: `1px ${spacing()}px`,
    };
  }, [theme]);

  return (
    <Container
      backgroundColor={colorMap[color].container}
      borderRadius={constraints.borderRadius}
      className={classes}
      data-type={type}
      fontFamily={fontMap.fontFamily}
      fontSize={fontMap.fontSize}
      fontWeight={fontMap.fontWeight}
      lineHeight={fontMap.lineHeight}
      padding={constraints.padding}
      ref={ref}
      style={style}
      textColor={colorMap[color].text}
      {...rest}
    >
      {children}
    </Container>
  );
});

/**
 * @todo Separate classes for `ContextualBadge.tsx` file
 */
interface IContainerProps {
  backgroundColor: string;
  textColor: string;
  fontFamily: string;
  fontSize: string | number;
  fontWeight: string | number;
  lineHeight: number;
  padding: string;
  borderRadius: number;
}

const Container = styled.span<IContainerProps>`
  background-color: ${(props) => props.backgroundColor};
  border-radius: ${(props) => props.borderRadius}px;
  color: ${(props) => props.textColor};
  font-family: ${(props) => props.fontFamily};
  font-size: ${(props) => props.fontSize};
  font-weight: ${(props) => props.fontWeight};
  line-height: ${(props) => props.lineHeight}px;
  padding: ${(props) => props.padding};
`;

export default withTheme(ContextualBadge);
