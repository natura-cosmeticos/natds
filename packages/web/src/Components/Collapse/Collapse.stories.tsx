import { Meta, Story } from '@storybook/react'
import * as React from 'react'
import Collapse, { ICollapseProps } from './Collapse'
import { List } from '../List'
import { ForNesting } from '../List/List.stories'

export default {
  component: Collapse,
  title: 'Utilities/Collapse'
} as Meta

const Template : Story<ICollapseProps> = (args: ICollapseProps) => <Collapse {...args} />

export const Playground : Story<ICollapseProps> = Template.bind({})
Playground.args = {
  children: 'Collapse content example'
}

export const WithList : Story<ICollapseProps> = Template.bind({})
WithList.args = {
  children: <List {...ForNesting.args} />,
  in: true,
  timeout: 'auto',
  unmountOnExit: true
}
