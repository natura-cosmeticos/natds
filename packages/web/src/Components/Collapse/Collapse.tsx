import * as React from 'react'
import MaterialCollapse from '@material-ui/core/Collapse'
import { ICollapseProps } from './Collapse.props'

export { ICollapseProps } from './Collapse.props'

export const Collapse = React.forwardRef<unknown, ICollapseProps>((
  props: ICollapseProps, ref
) => <MaterialCollapse {...props} ref={ref} />)

Collapse.displayName = 'Collapse'

export default Collapse
