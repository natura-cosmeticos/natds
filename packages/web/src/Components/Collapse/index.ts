import Collapse from './Collapse'

export { ICollapseProps } from './Collapse.props'
export { Collapse }
export default Collapse
