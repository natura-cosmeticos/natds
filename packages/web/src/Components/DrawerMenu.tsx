/* eslint-disable max-lines */
/* eslint-disable new-cap */
import React, { FunctionComponent, forwardRef } from "react";
import { makeStyles, withTheme } from "@material-ui/core";
import styled from "styled-components";
import { IThemeWeb } from "Themes";
import { iconNames } from "@naturacosmeticos/natds-icons";
import { getDefaultTheme } from "./shared";

import ContextualBadge from "./ContextualBadge";
import Collapse from "./Collapse";
import Icon from "./Icon";
import List from "./List";
import ListItem from "./ListItem";
import ListSubheader from "./ListSubheader";
import ListItemIcon from "./ListItemIcon";
import ListItemText from "./ListItemText";
import Typography from "./Typography";

const ICON_ARROW_DOWN = "outlined-navigation-arrowbottom";
const ICON_ARROW_UP = "outlined-navigation-arrowtop";

export interface IDrawerMenuProps {
  children?: React.ReactNode;
  component?: React.ElementType;
  list?: [IDrawerMenuSectionProps];
  theme: IThemeWeb | unknown;
}

/**
 * @todo Separate classes for `DrawerMenu.tsx` file
 */
export interface IDrawerMenuItemProps {
  name?: string;
  section?: string;
  icon?: keyof typeof iconNames;
  onSelect?: (event: React.SyntheticEvent, name?: string) => void;
  badge?: string;
  selected?: boolean;
  theme?: IThemeWeb | unknown;
}

export interface IDrawerMenuSectionProps extends IDrawerMenuItemProps {
  onToggle?: (event: React.SyntheticEvent, state: boolean, name?: string) => void;
  opened?: boolean;
  list?: [IDrawerMenuSectionProps];
}

// @todo refactor(web): refactor DrawerMenu component
export const DrawerMenu: FunctionComponent<IDrawerMenuProps> = forwardRef((
  props: IDrawerMenuProps,
  ref: any,
) => {
  const {
    children, list, theme: providerTheme, component, ...rest
  } = props;
  const theme: any = React.useMemo(() => getDefaultTheme(providerTheme), [providerTheme]);
  const StyledList = React.useMemo(() => makeStyles({
    padding: {
      marginBottom: theme.sizes.tiny,
      marginTop: theme.sizes.tiny,
      paddingBottom: 0,
      paddingTop: 0,
    },
  }), [theme]);

  const content = children || <List
    classes={StyledList()}
    dense
  >{list && list.map(BuildDrawerMenuItems(theme))}</List>;

  return (
    <DrawerMenuComponent {...rest} as={component} ref={ref}>
      {content}
    </DrawerMenuComponent>
  );
});

export const DrawerMenuSection: FunctionComponent<IDrawerMenuSectionProps> = forwardRef((
  props: IDrawerMenuSectionProps,
  ref: any,
) => {
  const {
    onToggle, icon, name, list, theme: providerTheme,
  } = props;
  const theme: any = React.useMemo(() => getDefaultTheme(providerTheme), [providerTheme]);
  const [
    opened, toggleSubmenu,
  ] = React.useState(props.opened);

  const StyledSubList = React.useMemo(() => makeStyles({
    padding: {
      borderLeft: `${theme.palette.action.hover} 1px solid`,
      margin: `0 0 0 ${theme.sizes.semix}px`,
      padding: `0 0 0 ${theme.sizes.tiny}px`,
    },
  }), [theme]);

  const StyledListSectionItem = React.useMemo(() => makeStyles({
    root: {
      borderRadius: theme.shape.borderRadius,
      cursor: "pointer",
      marginBottom: theme.sizes.micro,
      padding: theme.sizes.tiny,
      // eslint-disable-next-line sort-keys
      "&:hover": {
        backgroundColor: theme.palette.action.hover,
      },
      // eslint-disable-next-line sort-keys
      "&$selected": {
        backgroundColor: `${theme.palette.action.hover}`,
        // eslint-disable-next-line sort-keys
        "&:hover": {
          backgroundColor: theme.palette.action.hover,
        },
      },
    },
    selected: {},
  }), [theme]);

  const StyledListItemIcon = React.useMemo(() => makeStyles({
    root: {
      marginRight: theme.sizes.standard,
      marginTop: 0,
      minWidth: "auto",
    },
  }), [theme]);

  const classes = StyledSubList();
  const classesItem = StyledListSectionItem();
  const classesIcon = StyledListItemIcon();
  const listIconName = opened ? ICON_ARROW_UP : ICON_ARROW_DOWN;

  const handleClick = (event: React.SyntheticEvent) => {
    toggleSubmenu(!opened);

    /**
     * @todo refactor(web): convert expression to if block, add test
     */
    // eslint-disable-next-line no-unused-expressions
    onToggle && onToggle(event, !opened, name);
  };

  return (
    <>
      <ListItem
        onClick={handleClick}
        classes={classesItem}
        selected={opened}
        component="li"
        button
        ref={ref}>
        {icon && <ListItemIcon classes={classesIcon}><Icon name={icon} size="tiny" /></ListItemIcon>}
        <ListItemText>{getMenuItemText(props)}</ListItemText>
        <Icon name={listIconName} size="tiny" />
      </ListItem>
      <Collapse in={opened} timeout="auto" unmountOnExit component="li">
        {list && <List
          classes={classes}
          dense
        >{list.map(BuildDrawerMenuItems(theme))}</List>}
      </Collapse>
    </>
  );
});

// eslint-disable-next-line max-statements
export const DrawerMenuItem: FunctionComponent<IDrawerMenuItemProps> = forwardRef((
  props: IDrawerMenuItemProps,
  ref: any,
) => {
  const {
    onSelect, name, icon, selected, section, theme: providerTheme,
  } = props;
  const theme: any = React.useMemo(() => getDefaultTheme(providerTheme), [providerTheme]);

  const StyledListItem = React.useMemo(() => makeStyles({
    root: {
      borderRadius: theme.shape.borderRadius,
      cursor: "pointer",
      padding: theme.sizes.tiny,
      // eslint-disable-next-line sort-keys
      "&:hover": {
        backgroundColor: theme.palette.action.hover,
      },
    },
  }), [theme]);

  const StyledListSubheader = React.useMemo(() => makeStyles({
    root: {
      backgroundColor: `${theme.palette.background.paper}`,
      borderTop: `${theme.palette.action.hover} 1px solid`,
      margin: `${theme.sizes.small}px -${theme.sizes.tiny}px 0`,
    },
  }), [theme]);

  const StyledListItemIcon = React.useMemo(() => makeStyles({
    root: {
      marginRight: theme.sizes.standard,
      marginTop: 0,
      minWidth: "auto",
    },
  }), [theme]);

  if (section) {
    const classesSubheader = StyledListSubheader(theme);

    return (
      <ListSubheader classes={classesSubheader}><Typography variant="caption" component="span">{section}</Typography></ListSubheader>
    );
  }

  const classes = StyledListItem();
  const classesIcon = StyledListItemIcon();
  const handleClick = (event: React.SyntheticEvent) => onSelect && onSelect(event, name);
  const onClick = onSelect ? { onClick: handleClick } : {};

  return (
    <ListItem
      {...onClick}
      classes={classes}
      selected={selected}
      component="li"
      button
      ref={ref}>
      {icon && <ListItemIcon classes={classesIcon}><Icon name={icon} size="tiny" /></ListItemIcon>}
      <ListItemText>{getMenuItemText(props)}</ListItemText>
    </ListItem>
  );
});

const BuildDrawerMenuItems = (theme: IThemeWeb | unknown) => (props: IDrawerMenuSectionProps, key: number) => {
  if (props.list) {
    return <DrawerMenuSection theme={theme} key={key} {...props} />;
  }

  return <DrawerMenuItem theme={theme} key={key} {...props} />;

};

const getMenuItemText = (props: IDrawerMenuSectionProps) => {
  const { badge, name, theme } = props;

  if (!badge) {
    return name;
  }

  return (
    <>
      <Typography variant="body2" component="span">{name}</Typography>
      <StyledBadge as={ContextualBadge} theme={theme} color="error">{badge}</StyledBadge>
    </>
  );
};

const DrawerMenuComponent = styled.div`
  flex: 1 1 auto;
`;

const StyledBadge = styled.div<{ theme: IThemeWeb }>`
  margin-left: ${({ theme }) => `${theme.sizes.micro}px`};
`;

export default withTheme(DrawerMenu);
