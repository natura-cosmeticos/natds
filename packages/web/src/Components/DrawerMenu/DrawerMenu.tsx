// eslint-disable-next-line no-use-before-define
import * as React from 'react'
import { List } from '../List'
import { IDrawerMenuProps } from './DrawerMenu.props'
import { BuildDrawerMenuItems } from './BuildDrawerMenuItems'
import { useStyles } from './DrawerMenu.styles'

/**
 * ## Importing
 *
 * ```
 * import { DrawerMenu } from '@naturacosmeticos/natds-web';
 * ```
 */
export const DrawerMenu = React.forwardRef<HTMLElement, IDrawerMenuProps>((
  props: IDrawerMenuProps, ref
) => {
  const {
    children,
    component: Component = 'div',
    list,
    ...otherProps
  } = props

  const { root, listPadding } = useStyles()

  const content = children || (
  <List
    classes={{ padding: listPadding }}
    dense
  >
    {list && list.map((drawerMenuSectionProps, index) => (
      <BuildDrawerMenuItems {...drawerMenuSectionProps} key={index} />
    ))}
  </List>
  )

  return (
    <Component {...otherProps} className={root} ref={ref}>
      {content}
    </Component>
  )
})

DrawerMenu.displayName = 'DrawerMenu'

export default DrawerMenu
