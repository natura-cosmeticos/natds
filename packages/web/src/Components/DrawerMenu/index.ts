import DrawerMenu from './DrawerMenu'
import { DrawerMenuItem } from './DrawerMenuItem'
import { DrawerMenuSection } from './DrawerMenuSection'
import { MenuItemText } from './MenuItemText'
import { BuildDrawerMenuItems } from './BuildDrawerMenuItems'

export { IDrawerMenuProps } from './DrawerMenu.props'
export {
  DrawerMenu, DrawerMenuItem, DrawerMenuSection, MenuItemText, BuildDrawerMenuItems
}
export default DrawerMenu
