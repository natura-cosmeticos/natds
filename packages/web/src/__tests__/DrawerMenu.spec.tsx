import * as React from "react";
import * as renderer from "react-test-renderer";

import { expect } from "chai";
import { assert, spy } from "sinon";
import { mount } from "enzyme";

import DrawerMenu, { DrawerMenuItem, DrawerMenuSection, IDrawerMenuSectionProps } from "../Components/DrawerMenu";
import ContextualBadge from "../Components/ContextualBadge";
import Collapse from "../Components/Collapse";

const listBasic = [
  {
    icon: "outlined-action-love",
    name: "Menu Item",
  },
];

const listComplete = [
  {
    icon: "outlined-action-love",
    name: "Menu Item",
  },
  {
    badge: "Novo",
    icon: "outlined-action-love",
    name: "Menu Item",
    onSelect: () => null,
  },
  {
    badge: "Novo",
    icon: "outlined-action-love",
    list: [
      {
        name: "Menu Item",
        icon: "outlined-action-love",
      },
      {
        name: "Menu Item",
        icon: "outlined-action-love",
        badge: "New",
      },
    ],
    name: "Menu Item",
    onToggle: () => null,
    opened: false,
  },
  {
    section: "section",
  },
  {
    name: "Menu Item",
    icon: "outlined-action-love",
    badge: "Novo",
  },
];

describe("DrawerMenu component", () => {
  test("should match snapshot - DrawerMenu list", () => {
    const component = renderer.create(<DrawerMenu list={listComplete as unknown as [IDrawerMenuSectionProps]} />).toJSON();

    expect(component).matchSnapshot("DrawerMenu list snapshot");
  });

  test("should match snapshot - DrawerMenu custom component", () => {
    const component = renderer.create(<DrawerMenu list={listComplete as unknown as [IDrawerMenuSectionProps]} component="section" />).toJSON();

    expect(component).matchSnapshot("DrawerMenu custom component snapshot");
  });

  test("should match snapshot - DrawerMenu custom children", () => {
    const children = <div>some custom children</div>;
    const component = renderer.create(<DrawerMenu list={listComplete as unknown as [IDrawerMenuSectionProps]}>{children}</DrawerMenu>).toJSON();

    expect(component).matchSnapshot("DrawerMenu custom children snapshot");
  });

  test("should render custom children", () => {
    const children = <div className="custom">some custom children</div>;
    const component = mount(<DrawerMenu list={listBasic as [IDrawerMenuSectionProps]}>{children}</DrawerMenu>);

    expect(component).to.have.descendants(".custom");
    expect(component).to.not.have.descendants(DrawerMenuItem);
    expect(component).to.not.have.descendants(DrawerMenuSection);
  });

  test("should render list item", () => {
    const component = mount(<DrawerMenu list={listBasic as [IDrawerMenuSectionProps]} />);

    expect(component).to.have.descendants(DrawerMenuItem);
    expect(component).to.not.have.descendants(ContextualBadge);
    expect(component).to.not.have.descendants(DrawerMenuSection);
  });

  test("should render list item with badge", () => {
    const [listItem] = listBasic;
    const list = [
      {
        ...listItem,
        badge: "badge",
      },
    ];
    const component = mount(<DrawerMenu list={list as [IDrawerMenuSectionProps]} />);

    expect(component).to.have.descendants(ContextualBadge);
    expect(component).to.not.have.descendants(DrawerMenuSection);
  });

  test("should render submenu item", () => {
    const component = mount(<DrawerMenu list={listComplete as unknown as [IDrawerMenuSectionProps]} />);

    expect(component).to.have.descendants(DrawerMenuSection);
  });

  const INDEX_ONE = 1;
  const INDEX_TWO = 2;

  test("should start with submenu opened", () => {
    const listItem = listComplete[INDEX_TWO];
    const list = [
      {
        ...listItem,
        opened: true,
      },
    ];
    const component = mount(<DrawerMenu list={list as unknown as [IDrawerMenuSectionProps]} />);

    expect(component.find(Collapse).prop("in")).to.equal(true);
  });

  test("should open submenu and call onToggle", () => {
    const onToggle = spy(listComplete[INDEX_TWO], "onToggle");
    const component = mount(<DrawerMenu list={listComplete as unknown as [IDrawerMenuSectionProps]} />);

    expect(component.find(Collapse).prop("in")).to.equal(false);

    component.find("li").at(INDEX_TWO)
      .simulate("click");

    expect(component.find(Collapse).prop("in")).to.equal(true);

    assert.calledOnce(onToggle);
  });

  test("should select a item and call onSelect", () => {
    const onSelect = spy(listComplete[INDEX_ONE], "onSelect");
    const component = mount(<DrawerMenu list={listComplete as unknown as [IDrawerMenuSectionProps]} />);

    component.find("li").at(INDEX_ONE)
      .simulate("click");

    assert.calledOnce(onSelect);
  });
});
