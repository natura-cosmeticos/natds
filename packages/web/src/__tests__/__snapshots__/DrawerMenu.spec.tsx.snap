// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerMenu component should match snapshot - DrawerMenu custom children: DrawerMenu custom children snapshot 1`] = `
<div
  className="sc-AxiKw kCnJfj"
>
  <div>
    some custom children
  </div>
</div>
`;

exports[`DrawerMenu component should match snapshot - DrawerMenu custom component: DrawerMenu custom component snapshot 1`] = `
<section
  className="sc-AxiKw kCnJfj"
>
  <ul
    className="MuiList-root MuiList-dense MuiList-padding makeStyles-padding-13"
  >
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-14 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root makeStyles-root-15"
      >
        <i
          className="sc-AxirZ chGsOB natds-icons natds-icons-outlined-action-love undefined"
          size="tiny"
        />
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
        >
          Menu Item
        </span>
      </div>
      <span
        className="MuiTouchRipple-root"
      />
    </li>
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-16 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root makeStyles-root-17"
      >
        <i
          className="sc-AxirZ chGsOB natds-icons natds-icons-outlined-action-love undefined"
          size="tiny"
        />
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
        >
          <span
            className="MuiTypography-root MuiTypography-body2"
          >
            Menu Item
          </span>
          <span
            className="sc-AxjAm jLtmCw sc-AxhCb kLnmMw"
            fontFamily="\\"Roboto\\", sans-serif"
            fontSize="0.75rem"
            fontWeight={400}
          >
            Novo
          </span>
        </span>
      </div>
      <span
        className="MuiTouchRipple-root"
      />
    </li>
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-19 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root makeStyles-root-21"
      >
        <i
          className="sc-AxirZ chGsOB natds-icons natds-icons-outlined-action-love undefined"
          size="tiny"
        />
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
        >
          <span
            className="MuiTypography-root MuiTypography-body2"
          >
            Menu Item
          </span>
          <span
            className="sc-AxjAm jLtmCw sc-AxhCb kLnmMw"
            fontFamily="\\"Roboto\\", sans-serif"
            fontSize="0.75rem"
            fontWeight={400}
          >
            Novo
          </span>
        </span>
      </div>
      <i
        className="sc-AxirZ chGsOB natds-icons natds-icons-outlined-navigation-arrowbottom undefined"
        size="tiny"
      />
      <span
        className="MuiTouchRipple-root"
      />
    </li>
    <li
      className="MuiListSubheader-root makeStyles-root-22 MuiListSubheader-sticky MuiListSubheader-gutters"
    >
      <span
        className="MuiTypography-root MuiTypography-caption"
      >
        section
      </span>
    </li>
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-23 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root makeStyles-root-24"
      >
        <i
          className="sc-AxirZ chGsOB natds-icons natds-icons-outlined-action-love undefined"
          size="tiny"
        />
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
        >
          <span
            className="MuiTypography-root MuiTypography-body2"
          >
            Menu Item
          </span>
          <span
            className="sc-AxjAm jLtmCw sc-AxhCb kLnmMw"
            fontFamily="\\"Roboto\\", sans-serif"
            fontSize="0.75rem"
            fontWeight={400}
          >
            Novo
          </span>
        </span>
      </div>
      <span
        className="MuiTouchRipple-root"
      />
    </li>
  </ul>
</section>
`;

exports[`DrawerMenu component should match snapshot - DrawerMenu list: DrawerMenu list snapshot 1`] = `
<div
  className="sc-AxiKw kCnJfj"
>
  <ul
    className="MuiList-root MuiList-dense MuiList-padding makeStyles-padding-1"
  >
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-2 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root makeStyles-root-3"
      >
        <i
          className="sc-AxirZ chGsOB natds-icons natds-icons-outlined-action-love undefined"
          size="tiny"
        />
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
        >
          Menu Item
        </span>
      </div>
    </li>
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-4 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root makeStyles-root-5"
      >
        <i
          className="sc-AxirZ chGsOB natds-icons natds-icons-outlined-action-love undefined"
          size="tiny"
        />
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
        >
          <span
            className="MuiTypography-root MuiTypography-body2"
          >
            Menu Item
          </span>
          <span
            className="sc-AxjAm jLtmCw sc-AxhCb kLnmMw"
            fontFamily="\\"Roboto\\", sans-serif"
            fontSize="0.75rem"
            fontWeight={400}
          >
            Novo
          </span>
        </span>
      </div>
    </li>
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-7 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root makeStyles-root-9"
      >
        <i
          className="sc-AxirZ chGsOB natds-icons natds-icons-outlined-action-love undefined"
          size="tiny"
        />
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
        >
          <span
            className="MuiTypography-root MuiTypography-body2"
          >
            Menu Item
          </span>
          <span
            className="sc-AxjAm jLtmCw sc-AxhCb kLnmMw"
            fontFamily="\\"Roboto\\", sans-serif"
            fontSize="0.75rem"
            fontWeight={400}
          >
            Novo
          </span>
        </span>
      </div>
      <i
        className="sc-AxirZ chGsOB natds-icons natds-icons-outlined-navigation-arrowbottom undefined"
        size="tiny"
      />
    </li>
    <li
      className="MuiListSubheader-root makeStyles-root-10 MuiListSubheader-sticky MuiListSubheader-gutters"
    >
      <span
        className="MuiTypography-root MuiTypography-caption"
      >
        section
      </span>
    </li>
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-11 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root makeStyles-root-12"
      >
        <i
          className="sc-AxirZ chGsOB natds-icons natds-icons-outlined-action-love undefined"
          size="tiny"
        />
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
        >
          <span
            className="MuiTypography-root MuiTypography-body2"
          >
            Menu Item
          </span>
          <span
            className="sc-AxjAm jLtmCw sc-AxhCb kLnmMw"
            fontFamily="\\"Roboto\\", sans-serif"
            fontSize="0.75rem"
            fontWeight={400}
          >
            Novo
          </span>
        </span>
      </div>
    </li>
  </ul>
</div>
`;
